syntax = "proto3";
package mUSV;

import "google/protobuf/timestamp.proto";

message RequestData {
	int32 tag_id = 1;
	bool request_waypoints = 2;
}

message SensorData {
	message Pose2D {
		float x = 1;
		float y = 2;
		float yaw = 3;
		int32 xpx = 4;
		int32 ypx = 5;
	}
	Pose2D pose = 1;
	repeated int32 obstacle_sensors = 2 [packed=true];
	repeated int32 target_sensors = 3 [packed=true];
	google.protobuf.Timestamp timestamp = 4;
	message Waypoint {
		float x = 1;
		float y = 2;
	}
	repeated Waypoint waypoints = 5;
	bool loop_waypoints = 6;
	message ClusterPoint {
		float range = 1;
		float heading = 2;
	}
	ClusterPoint clusterPoint = 7;
}
